app/_includes/plugins/request-response-transformer/templates.md

```diff
@@ -0,0 +1,90 @@
+You can use any of the current request headers, query parameters, and captured URI
+groups as templates to populate supported configuration fields.
+
+{% table %}
+columns:
+  - title: Type
+    key: type
+  - title: Template
+    key: template
+rows:
+  - type: Header
+    template: |
+      * `$(headers.<header-name>)`
+      * `$(headers["<header-name>"])`
+  - type: Query parameter
+    template: |
+      * `$(query_params.<query-param-name>)`
+      * `$(query_params["<query-param-name>"])`
+  - type: Captured URIs
+    template: |
+      * `$(uri_captures.<group-name>)`
+      * `$(uri_captures["<group-name>"])`
+  - type: Shared variables
+    template: |
+      * `$(shared.<variable-name>)`
+      * `$(shared["<variable-name>"])`
+    
+{% endtable %}
+
+To escape a template, wrap it inside quotes and pass inside another template.
+For example:
+
+```
+$('$(something_that_needs_to_escaped)')
+```
+
+{:.info}
+> **Note:** The plugin creates a non-mutable table of request headers, query strings, and captured URIs before transformation. Therefore, any update or removal of parameters used in a template does not affect the rendered value of a template.
+
+### Advanced templates
+
+The content of the placeholder `$(...)` is evaluated as a Lua expression, so
+logical operators may be used. For example:
+
+```
+$(uri_captures["user-id"] or query_params["user"] or "unknown")
+```
+
+This will first look for the path parameter named `user-id` (in `uri_captures`). If not found, it will
+return the query parameter named `user`. If that also doesn't exist, it returns the default
+value '"unknown"'.
```

Comment:
```suggestion
This example will look for parameters in the following order:
1. Looks for a path parameter named `user-id` in `uri_captures`.
2. If not found, it will return the query parameter named `user`.
3. If that also doesn't exist, it returns the default value `"unknown"`.
```